Symfony/CS/Fixer/Contrib/Php4ConstructorFixer.php

```diff
@@ -0,0 +1,305 @@
+            array(T_OBJECT_OPERATOR),
+            array(T_STRING, $method),
+            '(',
+        );
+        $case = array(true, true, true, false, true);
+
+        // parse method parameters, if any
+        $index = $startIdx;
+        while (true) {
+            // find the next variable name
+            $index = $tokens->getNextTokenOfKind($index, array(array(T_VARIABLE)));
+
+            if (null === $index || $index >= $bodyIdx) {
+                // we've reached the body already
+                break;
+            }
+
+            // append a comma if it's not the first variable
+            if (count($seq) > 5) {
+                $seq[] = ',';
+                $case[] = true;
+            }
+
+            // append variable name to the sequence
+            $seq[] = array(T_VARIABLE, $tokens[$index]->getContent());
+            $case[] = true;
+        }
+
+        // almost done, close the sequence with ); }
+        $seq[] = ')';
+        $case[] = true;
+        $seq[] = ';';
+        $case[] = true;
+        $seq[] = '}';
+        $case[] = true;
+
+        return array($seq, $case);
+    }
+
+    /**
+     * Find a function or method matching a given name within certain bounds.
+     *
+     * @param Tokens $tokens   the Tokens instance
+     * @param string $name     the function/Method name
+     * @param int    $startIdx the function/method start index
+     * @param int    $bodyIdx  the function/method body index
+     *
+     * @return array|null {
+     *                    The function/method data, if a match is found.
+     *
+     *      @var int    $nameIdx  the index of the function/method name
+     *      @var int    $startIdx the index of the function/method start
+     *      @var int    $endInx   the index of the function/method end
+     *      @var string $bodyIdx  the index of the function/method body
+     * }
+     */
+    private function findFunction(Tokens $tokens, $name, $startIdx, $endIdx)
+    {
+        $function = $tokens->findSequence(array(
+            array(T_FUNCTION),
+            array(T_STRING, $name),
+        ), $startIdx, $endIdx, false);
+
+        if (null !== $function) {
+            // keep only the indexes
+            $function = array_keys($function);
+
+            // find previous block, ignoring method modifiers
+            $prevBlock = $tokens->getPrevMeaningfulToken($function[0]);
+            while ($tokens[$prevBlock]->isGivenKind(array(T_PUBLIC, T_PRIVATE, T_PROTECTED, T_ABSTRACT, T_STATIC))) {
+                if ($tokens[$prevBlock]->equals(array(T_ABSTRACT))) {
+                    // skip abstract methods
+                    return;
+                }
+                $prevBlock = $tokens->getPrevMeaningfulToken($prevBlock);
+            }
+
+            // find method body start
+            $bodyStart = $tokens->getNextTokenOfKind($function[1], array('{'));
+            if (null === $bodyStart) {
+                // no body
+                return;
```

Comment:
I've removed this for now. The check was meant to deal with potentially broken input files, but ther are surely other places where specifically crafted (broken) files could cause unexpected behaviour. My question is, what should a Fixer do in that case?